bundle/src/main/java/com/adobe/acs/commons/httpcache/store/jcr/impl/writer/BucketNodeFactory.java

```diff
@@ -0,0 +1,65 @@
+package com.adobe.acs.commons.httpcache.store.jcr.impl.writer;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jackrabbit.JcrConstants;
+import org.apache.jackrabbit.commons.JcrUtils;
+
+import com.adobe.acs.commons.httpcache.keys.CacheKey;
+import com.adobe.acs.commons.httpcache.store.jcr.impl.exceptions.BucketNodeFactoryException;
+
+/**
+ * Wrapper class for the bucket node
+ */
+public class BucketNodeFactory
+{
+    public static final double HASHCODE_LENGTH = 10d;
+
+    private final CacheKey key;
+    private final int cacheKeySplitDepth;
+    private final Node cacheRoot;
+
+    public BucketNodeFactory(Session session, String cacheRootPath, CacheKey key, int cacheKeySplitDepth)
+            throws RepositoryException, BucketNodeFactoryException
+    {
+        this.key = key;
+        this.cacheKeySplitDepth = cacheKeySplitDepth;
+
+        if(!session.nodeExists(cacheRootPath))
+            throw new BucketNodeFactoryException("Cache root path " + cacheRootPath + " not found!");
+        this.cacheRoot = session.getNode(cacheRootPath);
+    }
+
+    public Node getBucketNode() throws RepositoryException
+    {
+        final String[] pathArray = getPathArray();
+
+        Node targetNode = cacheRoot;
+
+        for(String path : pathArray){
+            Node childNode = JcrUtils.getOrCreateByPath(targetNode, path,false,"sling:Folder", "sling:Folder",false);
```

Comment:
`sling:Folder` is available via https://sling.apache.org/apidocs/sling8/org/apache/sling/jcr/resource/JcrResourceConstants.html